docs/config.rst

```diff
@@ -353,13 +348,83 @@ Or, if the option needs to be available as a function parameter::
         return (conf.some_setting if val is None else val) + 2
 
 
-See Also
-========
+.. _config-affiliated:
+
+Affiliated Packages
+===================
+
+If you maintain an affiliated package you can customise the behaviour of the
+astropy configuration machinery for your package. Prior to version 4.0 of
+Astropy the bahviour of both ``astropy`` and the package template was to write a
+configuration file to ``~/.astropy/yourpackage.cfg`` if a template configuration
+file was provided by your package. If you wish to maintain this behaviour you do
+not have to change anything.
+
+If you want to change the behaviour of your package to mimic that of ``astropy``
+you can define a wrapper for `astropy.config.write_default_config` in your
+``__init__.py`` file i.e.::
+
+
+  def write_default_config():
+      """
+      Writes out the template configuration file for this version of packagename.
+
+      This function will save a template config file for manual editing, if a
+      config file already exists, this will write a config file appended with the
+      version number, to facilitate comparison of changes.
+
+      Returns
+      -------
+      filepath : `pathlib.Path` or `None`
+          The full path of the file written or `None` if no file was written.
+      """
+      from . import config as _config
+      return _config.write_default_config('packagename')
+
+
+This function provides a way for your users to write out a version of your
+configuration template, in the same way that they are written on import before
+this change.
+
 
-.. toctree::
-   :maxdepth: 2
+Customizing the Configuration Directory
+---------------------------------------
 
-   config_0_4_transition
+.. warning::
+
+   If you used the :ref:`package template <packagetemplate:package-template>` to
+   create your affiliated package this might cause you to have to be more
```

Comment:
This has to be reworded with clarification. What exactly the affected package has to be careful about? What signs of trouble should they be looking out for, etc?